src/geometry/manifold/derivation_bundle.lean

```diff
@@ -0,0 +1,163 @@
+    (point_derivation.is_scalar_tower I M x)
+
+/-- The total bundle of point derivations. -/
+def derivation_bundle := Σ x : M, point_derivation I M x
+
+variables {I M}
+
+/-- The inclusion map of derivations at a point into the total bundle. -/
+def derivation_inclusion {x : M} (v : point_derivation I M x) : derivation_bundle I M :=
+sigma.mk x v
+
+instance [inhabited M] : inhabited (derivation_bundle I M) :=
+⟨derivation_inclusion (0 : point_derivation I M (default M))⟩
+
+section
+
+/- Why do I need to rewrite extensionality rules for reducible defs? -/
+namespace point_derivation
+
+variables {I} {M} {x y : M} {v w : point_derivation I M x} (f g : C^∞⟮I, M; 𝕜⟯) (r : 𝕜)
+
+lemma coe_injective (h : ⇑v = w) : v = w :=
+@derivation.coe_injective 𝕜 _ C^∞⟮I, M; 𝕜⟯ _ _ 𝕜 _ (point_derivation.module I M x) _
+  (point_derivation.is_scalar_tower I M x) v w h
+
+@[ext] theorem ext (h : ∀ f, v f = w f) : v = w :=
+coe_injective $ funext h
+
+theorem hext {u : point_derivation I M y} (h1 : x = y) (h2 : ∀ f, v f = u f) : v == u :=
+by { cases h1, rw heq_iff_eq, ext, exact h2 f }
+
+end point_derivation
+
+end
+
+section
+
+variables {I} {M} (X Y : derivation 𝕜 C^∞⟮I, M; 𝕜⟯ C^⊤⟮I, M; 𝕜⟯)
+  (f g : C^∞⟮I, M; 𝕜⟯) (r : 𝕜)
+
+namespace derivation
+
+/-- Evaluation of a global derivation at a point, giving a point derivation in the most natural
+possible way. -/
+def eval_map (x : M) : point_derivation I M x :=
+{ to_fun := λ f, (X f) x,
+  map_add' := λ f g, by { rw derivation.map_add, refl },
+  map_smul' := λ f g, by { rw [derivation.map_smul, algebra.id.smul_eq_mul], refl },
+  leibniz' := λ h k, by { dsimp only [], rw [derivation.leibniz, algebra.id.smul_eq_mul], refl } }
+
+/-- The evaluation is a linear map. -/
+def eval_at (x : M) : (derivation 𝕜 C^∞⟮I, M; 𝕜⟯ C^⊤⟮I, M; 𝕜⟯) →ₗ[𝕜] point_derivation I M x :=
+{ to_fun := λ X, X.eval_map x,
+  map_add' := λ X Y, rfl,
+  map_smul' := λ k X, rfl }
+
+lemma eval_apply (x : M) : eval_at x X f = (X f) x := rfl
+
+end derivation
+
+variables {E' : Type*} [normed_group E'] [normed_space 𝕜 E']
+{H' : Type*} [topological_space H'] {I' : model_with_corners 𝕜 E' H'}
+{M' : Type*} [topological_space M'] [charted_space H' M']
+
+/-- The differential of a function interpreted in the context of derivations. -/
+def fdifferential_map (f : C^∞⟮I, M; I', M'⟯) (x : M) (v : point_derivation I M x) :
+  (point_derivation I' M' (f x)) :=
+{ to_fun := λ g, v (g.comp f),
+  map_add' := λ g h, by rw [smooth_map.add_comp, derivation.map_add],
+  map_smul' := λ k g, by rw [smooth_map.smul_comp, derivation.map_smul],
+  leibniz' := λ g h, by {simp only [derivation.leibniz, smooth_map.mul_comp], refl} }
+
+/-- The differential is a linear map. -/
+def fdifferential (f : C^∞⟮I, M; I', M'⟯) (x : M) : (point_derivation I M x) →ₗ[𝕜]
+  (point_derivation I' M' (f x)) :=
+{ to_fun := fdifferential_map f x,
+  map_smul' := λ k v, rfl,
+  map_add' := λ v w, rfl }
+
+localized "notation `fd` := fdifferential" in manifold
```

Comment:
I am not a fan of incomprehensible notation -- is this notation standard amongst mathematicians? Can't we just call it `differential`?

Reply:
The standard notation is just a `d`, `fd` stands for frechet differential... I am not a fan of long names for things that will appear multiple times in long equations, but maybe I can use some special d! I am sure everybody who works with differential geometry will recognize a d as the differential